### PR TITLE
Add information about cache folder outside of site root

### PR DESCRIPTION
**Pull request by jadwigo**
Adds an explanation in docs for https://github.com/bolt/bolt/issues/7440

### CODE REVIEW

File: docs/howto/troubleshooting-outside-webroot.md [hunk 1]
```diff
@@ -261,6 +261,60 @@ line with the `require` in it, and change it like this:
...
+│       ├── app/
+│       ├── extensions/
+│       ├── public/
+│       └── vendor/
+└── shared/
+    ├── cache/
+    └── files/
+```
+
+The bolt composer update or installation will attempt to update the paths in `.bolt.yml`. if the
```
Comment (xiaohutai) [inline docs/howto/troubleshooting-outside-webroot.md]: `bolt` -> `Bolt`

File: docs/howto/troubleshooting-outside-webroot.md [hunk 2]
```diff
@@ -261,6 +261,60 @@ line with the `require` in it, and change it like this:
...
+cache path is absolute it will add an `app` path that will look like `app: %site%/../../shared/cache`
+or something similar. You have to manually set the `app` path in the `.bolt.yml` configuration file
+to `app: %site%/app` for bolt to work.
```
Comment (xiaohutai) [inline docs/howto/troubleshooting-outside-webroot.md]: `bolt` -> `Bolt`

**Comment by jadwigo** [comment]
The behaviour did occur on updates from 3.3.x to 3.4.10+ or updates from 3.3.x to 3.5.x so I assumed it was only relevant for bolt 3.4 / 3.5

As for the location in the docs for troubleshooting outside the webroot - that was a bit of a stretch. It was the closest match for where I would look. There is no "miscellaneous troubleshooting" or "advanced configuration with bolt.yml" document yet, so I put it here.

**Comment by GawainLynch** [comment]
> The behaviour did occur on updates from 3.3.x to 3.4.10+ or updates from 3.3.x to 3.5.x so I assumed it was only relevant for bolt 3.4 / 3.5

Yeah, I've been working around it since late 3.0.0-beta on my sites, and I mentioned that to @bobdenotter in Slack but it didn't make it to you.

> As for the location in the docs for troubleshooting outside the webroot - that was a bit of a stretch. It was the closest match for where I would look. There is no "miscellaneous troubleshooting" or "advanced configuration with bolt.yml" document yet, so I put it here.

Yeah, I have to think through it myself. Thing is that it doesn't just affect cache, but also the database directory is another one that will catch you out if you're using Sqlite.

**Comment by jadwigo** [comment]
How about moving the new section to `configuration/troubleshooting-configuration.md`

**Comment by GawainLynch** [comment]
I was thinking `docs/installation/composer-create-project/creating-a-new-project.md` as that is where `.bolt.yml` is documented

**Comment by jadwigo** [comment]
For me `.bolt.yml` is used in more than just projects created with composer - I need it in sites managed with deployer too.

**Comment by GawainLynch** [comment]
That is an implementation detail for you though.

**Comment by jadwigo** [comment]
It does not mean that I'm set on keeping the page where I put it... feel free to move and modify as needed.

**Comment by JarJak** [comment]
Docs in this PR are too exceptional. It should cover more cases. Example:
Before:
```
paths:
    cache: %site%/var/cache
    database: %site%/var/database
```
After:
```
paths:
    app: '%site%/%site%/var'
```

So this is wrong:
>If the cache path is absolute it will add an `app` path that will look like `app: %site%/../../shared/cache` or similar.
1. It will not only add `app` but also remove other paths from `.bolt.yml`
2. This happens not only with absolute paths.

**Comment by jadwigo** [comment]
Oh sorry... I deleted the branch, not realizing it was still in use.

**Comment by JarJak** [comment]
@gawainlynch can you reopen this?

**Comment by GawainLynch** [comment]
Unfortunately no, as the branch is no longer so GitHub greys out the reopen button. But also the details were a long way off anyway, and there would be less effort just using the diff for inspiration and starting fresh :innocent:

**Comment by jadwigo** [comment]
@JarJak if you have the time, please make a new pull request for the docs, I'm unable to at the moment

**Pull request closed without merging**